File: leetcode/hash-map-set/cpp/1657.determine-if-two-strings-are-close.cpp
/*
 * @lc app=leetcode id=1657 lang=cpp
 *
 * [1657] Determine if Two Strings Are Close
 */

// @lc code=start
class Solution {
public:
	bool closeStrings(string word1, string word2) {
		if (word1.size() != word2.size()) return false;

		vector<int>	v1(26, 0), v2(26,0);
		for (char c : word1) v1[c - 'a']++;
		for (char c : word2) v2[c - 'a']++;
		for (int i = 0; i < 26; i++) {
			if ((v1[i] == 0 && v2[i] > 0) || (v2[i] == 0 && v1[i] > 0))
				return false;
		}
		sort(v1.begin(), v1.end());
		sort(v2.begin(), v2.end());
		return v1 == v2;
	}
};
// @lc code=end

File: leetcode/hash-map-set/py/1657.determine-if-two-strings-are-close.py
#
# @lc app=leetcode id=1657 lang=python3
#
# [1657] Determine if Two Strings Are Close
#

# @lc code=start
class Solution:
	def closeStrings(self, word1: str, word2: str) -> bool:
		if len(word1) != len(word2): return False
		s1 = set(word1)
		if s1 != set(word2): return False
		return sorted([word1.count(c) for c in s1]) == sorted([word2.count(c) for c in s1])

# @lc code=end
